analyses/npl/09_flood_events_by_municipality.md

```diff
@@ -0,0 +1,300 @@
+VERSION_LOC = "_v3"
+
+KARNALI_BASINS = [
+    'Karnali', 'West Rapti', 'Babai'
+]
+KOSHI_WATERSHED = 'Saptakoshi'
+
+DATA_DIR = Path(os.environ["AA_DATA_DIR"]) 
+DATA_DIR_PRIVATE = DATA_DIR / 'private'
+DATA_DIR_PUBLIC = DATA_DIR / 'public'
+RCO_DIR = DATA_DIR_PRIVATE / 'exploration/npl/unrco'
+SHAPEFILE_DIR = DATA_DIR_PUBLIC / 'raw/npl/cod_ab'
+
+
+BASINS_SHAPEFILE = RCO_DIR / 'shapefiles/Major_River_Basins.shp'
+WATERSHED_SHAPEFILE = RCO_DIR / 'shapefiles/Major_watershed.shp'
+ADMIN_SHAPEFILE = SHAPEFILE_DIR / 'npl_admbnda_ocha_20201117/npl_admbnda_nd_20201117_shp.zip'
+#ADMIN2_SHAPEFILE = 'npl_admbnda_districts_nd_20201117.shp'
+ADMIN2_SHAPEFILE = 'npl_admbnda_adm2_nd_20201117.shp'
+
+PAST_EVENTS_FILENAME = RCO_DIR / 'HistoricalReportedIncident_Municipality Level_AAPilot.xlsx'
+
+GLOFAS_DIR = DATA_DIR / "public/exploration/npl/glofas"
+GLOFAS_RP_FILENAME = GLOFAS_DIR / "glofas_return_period_values.xlsx"
+
+MUNICIPALITIES_OUTPUT_GEOPACKAGE = RCO_DIR / 'shapefiles/municipalities_of_interest.gpkg'
+```
+
+### Read in and clean data
+
+Read in GloFAS data
+
+```python
+ds_glofas_reanalysis = utils.get_glofas_reanalysis(
+    country_iso3=COUNTRY_ISO3)
+df_return_period =  pd.read_excel(GLOFAS_RP_FILENAME, index_col='rp')
+```
+
+Read in events and clean
+
+```python
+df_events = (pd.read_excel(PAST_EVENTS_FILENAME, index_col='S.No.')
+             .rename(columns={'Mun_ETHOS_Code': 'pcode'})
+            )
+# Drop events with no date and convert date column
+df_events = df_events.loc[df_events['Year'] > 0]
+# For days that are 0, assume they mean the middle of the month
+df_events['Day'] = np.where(df_events['Day'] == 0, 15, df_events['Day'])
+df_events['Incident Date'] = pd.to_datetime(df_events[['Year', 'Month', 'Day']])
+
+# Only select events in the time range of GloFAS
+df_events = df_events.loc[df_events['Incident Date'] > ds_glofas_reanalysis.time[0].data]
+```
+
+Merge event data with admin
+
+```python
+df_admin = (gpd.read_file(f'zip://{ADMIN_SHAPEFILE}!{ADMIN2_SHAPEFILE}')
+            .rename(columns={'ADM2_PCODE': 'pcode'}))
+
+df_events = pd.merge(df_admin, df_events, how='right', left_on='pcode', right_on='pcode')
+```
+
+Read in Basin and Watershed to get areas of interest
+
+```python
+df_basins = gpd.read_file(BASINS_SHAPEFILE)
+df_watershed = gpd.read_file(WATERSHED_SHAPEFILE)
+```
+
+```python
+df_basins['Karnali'] = 0
+idx = df_basins['Major_Basi'].isin([f'{basin} River Basin' for basin in KARNALI_BASINS])
+df_basins.loc[idx, 'Karnali']= 1
+karnali = df_basins[idx].dissolve(by='Karnali')
+koshi = df_watershed[df_watershed['WSH_NME'] == f'{KOSHI_WATERSHED} Watershed']
+```
+
+Plot event municipalities over the areas of interest
+
+```python
+fig, ax = plt.subplots()
+karnali.plot(ax=ax)
+koshi.plot(ax=ax)
+df_events.boundary.plot(ax=ax, color='r', lw=0.5)
+```
+
+Might be easier to just do a latitude cut. Latitude > 86 is Koshi, < 83 is Karnali.
```

Comment:
Or a spatial join/intersect? Although this works well enough.

Reply:
I chose the latitude cut because a couple of municipalities in the easternmost group don't quite intersect with the Saptakoshi watershed.